fetch.bs

```diff
@@ -4618,21 +4618,93 @@ steps:
-     <var>length</var>), (`<code>Content-Type</code>`, <var>type</var>) », and
-     <a for=response>body</a> is <var>body</var>.
+     <li><p>If <var>response</var>'s <a for=response>range-requested flag</a> is unset, run the
+     following steps:
+
+     <ol>
+       <li><p>Set <var>response</var>'s <a for=response>status message</a> to `<code>OK</code>`
+
+       <li><p>Set <var>response</var>'s <a for=response>body</a> to <var>body</var>.
+
+       <li><p>Set <var>response</var>'s <a for=response>header list</a> to «
+       (`<code>Content-Length</code>`, <var>bodyLength</var>), (`<code>Content-Type</code>`,
+       <var>type</var>) »
+     </ol>
+
+     <li><p>If <var>response</var>'s <a for=response>range-requested flag</a> is set, run the
+     following steps:
+
+     <ol>
+       <li><p>Let <var>rangeHeader</var> be the result of <a>extracting header list values</a> given
+       `<code>Range</code>` and <var>request</var>'s <a for=response>header list</a>.
+
+       <li><p>Let <var>rangeValue</var> be the result of <a>parsing a single range header value</a>
+       given <var>rangeHeader</var>.
+
+       <li><p>If <var>rangeValue</var> is failure, then return a <a>network error</a>.
+
+       <li>
+         <p>blob.slice(<var>rangeStart</var>[0], <var>rangeValue</var>[1], <var>type</var>) here
+         on <var>body</var>.
+
+         <p class="note">TODO(dlrobertson): This requires a lot more thought.
+
+       <li>
+         <p>Let <var>contentRange</var> be `<code>bytes </code>`.
+
+         <p class="note">TODO(dlrobertson): Split this out into something like
+         <a>add a range header</a>
+
+       <li><p>If <var>rangeValue</var>[0] is not null, <a lt="serialize an integer">Serialize</a> and
+       <a>isomorphic encode</a> <var>rangeValue</var>[0], and append the result to
+       <var>contentRange</var>.
+
+       <li>
+         <p>If <var>rangeValue</var>[0] is null, <a lt="serialize an integer">Serialize</a> and
+         <a>isomorphic encode</a> <var>rangeValue</var>[1] - <var>body</var>'s
+         <a for=body>length</a>, and append the result to <var>contentRange</var>.
+
+         <p class="note">TODO(dlrobertson): This can probably be worded better. Maybe use a result
+         from blob.slice.
+
+       <li><p>Append 0x2D (-) to <var>contentRange</var>.
+
+       <li><p>If <var>rangeValue</var>[1] is not null, <a lt="serialize an integer">Serialize</a> and
+       <a>isomorphic encode</a> <var>rangeValue</var>[1], and append the result to
+       <var>contentRange</var>.
+
+       <li><p>If <var>rangeValue</var>[1] is null, append <var>bodyLength</var> to
+       <var>contentRange</var>
+
+       <li><p>Append 0x2F (/) to <var>contentRange</var>.
+
+       <li><p>Append <var>bodyLength</var> to <var>contentRange</var>.
+
+       <li><p>Set <var>response</var>'s <a for=response>status message</a> to
+       `<code>Partial Content</code>`
```

Comment:
You want to set status to 206 as well here. (It might be okay to not set status message at all and let it be the empty byte sequence as it would be in HTTP/2 and up, but I don't care strongly.)